tests/Doctrine/Tests/ORM/Functional/ValueConversionType/ManyToManyCompositeIdForeignKeyTest.php

```diff
@@ -0,0 +1,200 @@
+<?php
+
+namespace Doctrine\Tests\ORM\Functional\ValueConversionType;
+
+use Doctrine\DBAL\Types\Type as DBALType;
+use Doctrine\Tests\Models\ValueConversionType as Entity;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * The entities all use a custom type that converst the value as identifier(s).
+ * {@see \Doctrine\Tests\DbalTypes\Rot13Type}
+ *
+ * Test that ManyToMany associations with composite id of which one is a
+ * association itself work correctly.
+ *
+ * @group DDC-3380
+ */
+class ManyToManyCompositeIdForeignKeyTest extends OrmFunctionalTestCase
+{
+    public function setUp()
+    {
+        if (DBALType::hasType('rot13')) {
+            DBALType::overrideType('rot13', 'Doctrine\Tests\DbalTypes\Rot13Type');
+        } else {
+            DBALType::addType('rot13', 'Doctrine\Tests\DbalTypes\Rot13Type');
+        }
+
+        $this->useModelSet('vct_manytomany_compositeid_foreignkey');
+        parent::setUp();
+
+        $auxiliary = new Entity\AuxiliaryEntity();
+        $auxiliary->id4 = 'abc';
+
+        $inversed = new Entity\InversedManyToManyCompositeIdForeignKeyEntity();
+        $inversed->id1 = 'def';
+        $inversed->foreignEntity = $auxiliary;
+
+        $owning = new Entity\OwningManyToManyCompositeIdForeignKeyEntity();
+        $owning->id2 = 'ghi';
+
+        $inversed->associatedEntities->add($owning);
+        $owning->associatedEntities->add($inversed);
+
+        $this->_em->persist($auxiliary);
+        $this->_em->persist($inversed);
+        $this->_em->persist($owning);
+
+        $this->_em->flush();
+        $this->_em->clear();
+    }
+
+    public static function tearDownAfterClass()
```

Comment:
I did this to prevent issues running the entire test suite: Other tests don't have the custom mapping-type registered, but when the tables still exist I ran into exceptions stating Doctrine couldn't find out what mapping-type to use for the identifier columns, and the suite would break.